src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -628,6 +648,55 @@ private void processLogsAndUpload() throws InterruptedException {
         }
     }
 
+    /**
+     * calculate the hash for the active file, which takes and digests first some lines (default is 1) in the file.
+     * If the content of file is insufficient, or incorrect setting for DEFAULT_LINES_FOR_DIGEST_NUM, the hash is null.
+     * @param activeFile
+     * @return the nullable calculated hash for active file
+     */
+    private Optional<String> calculateHashForActiveFile(File activeFile) {
+        Optional activeFileHash = Optional.ofNullable(null);
+        // String activeFileHash = null;
+        try (SeekableByteChannel chan = Files.newByteChannel(activeFile.toPath(), StandardOpenOption.READ);
+             PositionTrackingBufferedReader r =
+                new PositionTrackingBufferedReader(new InputStreamReader(Channels.newInputStream(chan),
+                        StandardCharsets.UTF_8))) {
+            r.setInitialPosition(0);
+            chan.position(0);
+            StringBuilder data = new StringBuilder(r.readLine());
+
+            // if the lines for digest is less than 1 line, skip this file and generate error.
+            if (DEFAULT_LINES_FOR_DIGEST_NUM < 1) {
+                logger.atError().cause(new InvalidDigestTargetLinesNumberException("The active log file is " +
+                        "not read because the Digest requires at least one line message."));
+            } else {
```

Comment:
IMO: it is generally better not to nest conditional statement it makes the code harder to follow. A few ways to remedy this is to 

* Extract code into methods
* Use guard pattern [See](https://refactoring.guru/replace-nested-conditional-with-guard-clauses)